<a href="https://colab.research.google.com/github/VictorAlfa/ITLab_Python/blob/main/Avalia%C3%A7%C3%A3o_1_Proderj_24_08.ipynb" target="_parent"><img src="https://colab.research.google.com/assets/colab-badge.svg" alt="Open In Colab"/></a>

#NOMES:
* VICTOR AMADEU OLIVEIRA PIRES DOS SANTOS

# Avaliação



In [78]:
!pip install numpy  #instala a biblioteca Numpy
import numpy as np

#### Considere a seguinte lista:

In [79]:
lista_A = [2, 29, 8, 98, 10, 3, -4, 9, 10, 98]

#### Usando as funções apresentadas, faça os exercícios de 1 à 4:

1) Imprima a soma do maior e o menor elemento da lista:


In [80]:
soma_max_min = max(lista_A) + min(lista_A)
print(f'soma maior e menor é {soma_max_min}')

soma maior e menor é 94


2) Imprima o número de ocorrências do maior elemento da lista:

In [81]:
max_count = lista_A.count(max(lista_A))
print(f'o numero {max(lista_A)} aparece {max_count} vezes')

o numero 98 aparece 2 vezes


3) Imprima o maior número ímpar da lista utilizando um código para calculá-lo:

In [82]:
max_impar = max([num for num in lista_A if num%2 != 0])
print(f'max_impar é {max_impar}')

#max_impar = float()
#for num in lista_A:
#  if num %2 !=0 and num > max_impar:
#    max_impar = num
#    break

max_impar é 29


4) Imprima o maior elemento da lista elevado ao quadrado:

In [83]:
max_lista_A = max(lista_A)
print(f'max² é {max_lista_A**2}')

#max_max = np.square(max(lista_A))
#print(max_max)

max² é 9604


5) Crie uma lista com todos os números pares de 0 até 1000 (incluído).

In [84]:
lista_par = [num for num in range(0,1001) if num%2 == 0]
print(lista_par)

[0, 2, 4, 6, 8, 10, 12, 14, 16, 18, 20, 22, 24, 26, 28, 30, 32, 34, 36, 38, 40, 42, 44, 46, 48, 50, 52, 54, 56, 58, 60, 62, 64, 66, 68, 70, 72, 74, 76, 78, 80, 82, 84, 86, 88, 90, 92, 94, 96, 98, 100, 102, 104, 106, 108, 110, 112, 114, 116, 118, 120, 122, 124, 126, 128, 130, 132, 134, 136, 138, 140, 142, 144, 146, 148, 150, 152, 154, 156, 158, 160, 162, 164, 166, 168, 170, 172, 174, 176, 178, 180, 182, 184, 186, 188, 190, 192, 194, 196, 198, 200, 202, 204, 206, 208, 210, 212, 214, 216, 218, 220, 222, 224, 226, 228, 230, 232, 234, 236, 238, 240, 242, 244, 246, 248, 250, 252, 254, 256, 258, 260, 262, 264, 266, 268, 270, 272, 274, 276, 278, 280, 282, 284, 286, 288, 290, 292, 294, 296, 298, 300, 302, 304, 306, 308, 310, 312, 314, 316, 318, 320, 322, 324, 326, 328, 330, 332, 334, 336, 338, 340, 342, 344, 346, 348, 350, 352, 354, 356, 358, 360, 362, 364, 366, 368, 370, 372, 374, 376, 378, 380, 382, 384, 386, 388, 390, 392, 394, 396, 398, 400, 402, 404, 406, 408, 410, 412, 414, 416, 418, 420,

6) Crie uma lista com todos os números ímpares de 0 até 199 (incluído)

In [85]:
lista_impar = [num for num in range(0,200) if num%2 != 0]
print(lista_impar)

[1, 3, 5, 7, 9, 11, 13, 15, 17, 19, 21, 23, 25, 27, 29, 31, 33, 35, 37, 39, 41, 43, 45, 47, 49, 51, 53, 55, 57, 59, 61, 63, 65, 67, 69, 71, 73, 75, 77, 79, 81, 83, 85, 87, 89, 91, 93, 95, 97, 99, 101, 103, 105, 107, 109, 111, 113, 115, 117, 119, 121, 123, 125, 127, 129, 131, 133, 135, 137, 139, 141, 143, 145, 147, 149, 151, 153, 155, 157, 159, 161, 163, 165, 167, 169, 171, 173, 175, 177, 179, 181, 183, 185, 187, 189, 191, 193, 195, 197, 199]


7) Imprima o valor da média de todos os elementos da lista_A:

In [86]:
print(f'media lista_A é {sum(lista_A)/len(lista_A)}')

#media_lista = np.mean(lista_A)
#print(f'media lista_A é {media_lista}')

media lista_A é 26.3


8) Crie 3 tuplas para armazenar as notas dos alunos em uma disciplina. Cada tupla deverá conter o nome do aluno (escolhido por você) seguido de suas duas notas.

In [87]:
Aluno_1 = ("Roberto", 9.5, 3.0)
Aluno_2 = ("Francisco", 5.3, 7.0)
Aluno_3= ("Bernardo", 10.0, 9.0)

9) Imprima a soma das notas de cada aluno do exercício 8, use as variáveis criadas por você.
A resposta deve ter o seguinte formato:

O somatório das notas do {Aluno_1} é {nota do Aluno_1}

O somatório das notas do {Aluno_2} é {nota do Aluno_2}

O somatório das notas do {Aluno_3} é {nota do Aluno_3}

In [88]:
print(f'O somatório das notas do {Aluno_1[0]} é {sum(Aluno_1[1:])}')
print(f'O somatório das notas do {Aluno_2[0]} é {sum(Aluno_2[1:])}')
print(f'O somatório das notas do {Aluno_3[0]} é {sum(Aluno_3[1:])}')

O somatório das notas do Roberto é 12.5
O somatório das notas do Francisco é 12.3
O somatório das notas do Bernardo é 19.0


10) Imprima o nome e a maior nota de cada aluno do exercício 8, use as variáveis criadas por você.

A maior nota do {Aluno_1} é {maior_nota_do_Aluno_1} e a sua menor nota é {menor_nota_do_Aluno_1}

A maior nota do {Aluno_2} é {maior_nota_do_Aluno_2} e a sua menor nota é {menor_nota_do_Aluno_2}

A maior nota do {Aluno_3} é {maior_nota_do_Aluno_3} e a sua menor nota é {menor_nota_do_Aluno_3}

In [89]:
print(f'A maior nota do {Aluno_1[0]} é {max(Aluno_1[1:])} e a sua menor nota é {min(Aluno_1[1:])}')
print(f'A maior nota do {Aluno_2[0]} é {max(Aluno_2[1:])} e a sua menor nota é {min(Aluno_2[1:])}')
print(f'A maior nota do {Aluno_3[0]} é {max(Aluno_3[1:])} e a sua menor nota é {min(Aluno_3[1:])}')

A maior nota do Roberto é 9.5 e a sua menor nota é 3.0
A maior nota do Francisco é 7.0 e a sua menor nota é 5.3
A maior nota do Bernardo é 10.0 e a sua menor nota é 9.0


11) Construa um dicionário para armazenar os CNPJs de 5 empresas e o lucro ou prejuízo acumulado no ano de 2021. A chave deve ser o nome da empresa e o valor deve ser o cnpj.








* Petrobras
CNPJ 33.000.167/0661-29
faturamento 2021: R$ 452,7bilhões

* JBS
CNPJ 02.916.265/0001-60
faturamento 2021: R$ 350,7bilhões

* Vale
CNPJ 33.592.510/0001-54
faturamento 2021: R$ 293,5bilhões

* Raízen
CNPJ 08.070.508/0001-78
faturamento 2021: R$ 137,8bilhões

* Vibra
CNPJ 34.274.233/0001-02
faturamento 2021: R$ 130,1bilhões

In [90]:
# Dic Empresas CNPJ
empresas = {
    'Petrobras': '33.000.167/0661-29',
    'JBS': '02.916.265/0001-60',
    'Vale':  '33.592.510/0001-54',
    'Raízen': '08.070.508/0001-78',
    'Vibra': '34.274.233/0001-02'
}
print(empresas)

# Dic Empresas Faturamento
faturamentos_2021 = {
    'Petrobras':452.7,
    'JBS':350.7,
    'Vale':293.5,
    'Raízen':137.8,
    'Vibra': 130.1
}
print(faturamentos_2021)

{'Petrobras': '33.000.167/0661-29', 'JBS': '02.916.265/0001-60', 'Vale': '33.592.510/0001-54', 'Raízen': '08.070.508/0001-78', 'Vibra': '34.274.233/0001-02'}
{'Petrobras': 452.7, 'JBS': 350.7, 'Vale': 293.5, 'Raízen': 137.8, 'Vibra': 130.1}


In [108]:
# Lista Empresas
empresas = [
    ['Petrobras', '33.000.167/0661-29', 452.7],
    ['JBS', '02.916.265/0001-60', 350.7],
    ['Vale', '33.592.510/0001-54', 293.5],
    ['Raízen', '08.070.508/0001-78', 137.8],
    ['Vibra', '34.274.233/0001-02', 130.1]
]
print(empresas)

[['Petrobras', '33.000.167/0661-29', 452.7], ['JBS', '02.916.265/0001-60', 350.7], ['Vale', '33.592.510/0001-54', 293.5], ['Raízen', '08.070.508/0001-78', 137.8], ['Vibra', '34.274.233/0001-02', 130.1]]


12) Usando o dicionário criado no exercício 11, imprima o valor do menor e do maior faturamento e a média do faturamento das 5 empresas. A reposta deve ter o seguinte formato:

* A empresa com o maior faturamento {valor do faturamento} é {nome da empresa}.
* A empresa com o menor faturamento {valor do faturamento} é {nome da empresa}.
* A média do faturamento das cinco empresas em 2021 foi de {média do faturamento}.

In [92]:
#A empresa com o maior faturamento {valor do faturamento} é {nome da empresa}
max_faturamento_2021 = max(faturamentos_2021.values())
nome_max_faturamento = [nome for nome, faturamento in faturamentos_2021.items() if faturamento == max_faturamento_2021][0]

#A empresa com o menor faturamento {valor do faturamento} é {nome da empresa}
min_faturamento_2021 = min(faturamentos_2021.values())
nome_min_faturamento = [nome for nome, faturamento in faturamentos_2021.items() if faturamento == min_faturamento_2021][0]

#A média do faturamento das cinco empresas em 2021 foi de {média do faturamento}
media_faturamento_2021 = sum(faturamentos_2021.values()) / len(faturamentos_2021)

print(f'A empresa com o maior faturamento {max_faturamento_2021} é {nome_max_faturamento}')
print(f'A empresa com o menor faturamento {min_faturamento_2021} é {nome_min_faturamento}')
print(f'A média do faturamento das cinco empresas em 2021 foi de {media_faturamento_2021}')

A empresa com o maior faturamento 452.7 é Petrobras
A empresa com o menor faturamento 130.1 é Vibra
A média do faturamento das cinco empresas em 2021 foi de 272.96


13) Sabendo que A = {0, 1, 2, 3, 4, 5, 6}, B = {6, 7, 8, 9} e C = {2, 4, 6, 8, 10}, quais são os elementos do conjunto (A∩B)UC?

In [93]:
A = {0, 1, 2, 3, 4, 5, 6}
B = {6, 7, 8, 9}
C = {2, 4, 6, 8, 10}

conjunto = (A & B) | C
print(f'{conjunto}')

{2, 4, 6, 8, 10}


14) Os alunos de uma escola devem se inscrever em DUAS MODALIDADES esportivas no início do ano, porém, as inscrições são abertas e alguns alunos podem ter se inscrito em apenas uma modalidade ou em três. As modalidades esportivas são: NATAÇÃO, CICLISMO E CORRIDA. Você é o responsável pelas inscrições e deverá analisar as listagens com as inscrições e realizar as seguintes ações corretivas:

14.1. Para cada uma das três modalidades, verifique se o mesmo aluno se inscreveu mais de uma vez e exclua exclua os nomes duplicados.

14.2. Em seguida, contabilize a quantidade de alunos inscritos em
cada modalidade e imprima os três valores

14.3 Imprima uma listagem com o nome de todos os alunos nas 3 modalidades e a quantidade total de alunos, somadas as 3 modalidades.

14.4 Imprima uma listagem com o nome de cada aluno e as modalidades em que está inscrito

14.5 Imprima uma listagem com os alunos que se inscreveram em apenas 1 ou em 3 modalidades

In [94]:
natacao = ["Nicolas da Silva","Marcos Santos", "Luciana Barbosa", "Ana Schimidt", "Marcos Santos", "Joana Ramos", "André Consentino", "Marcela Rangel", "Luciana Barbosa", "Carlos Antunes"]
ciclismo = ["Nicolas da Silva", "Marcos Santos", "Luciana Barbosa", "Ana Schimidt", "Marcos Santos","Marcela Rangel", "Michele Souza", "Marcelo Bach", "Luis Pereira"]
corrida = ["Marcos Santos", "Luciana Barbosa", "Ana Schimidt", "Marcos Santos", "Luis Pereira", "Joana Ramos", "André Consentino","Marcela Rangel", "Luis Gomes", "Luciana Barbosa"]

In [95]:
#14.1. Para cada uma das três modalidades, verifique se o mesmo aluno se inscreveu mais de uma vez e exclua exclua os nomes duplicados

#verificando quem são os duplicados
def verificar_duplicados(lista):
    duplicados = {}
    for aluno in lista:
        if lista.count(aluno) > 1:
            duplicados[aluno] = lista.count(aluno)
    return duplicados

duplicados_natacao = verificar_duplicados(natacao)
duplicados_ciclismo = verificar_duplicados(ciclismo)
duplicados_corrida = verificar_duplicados(corrida)

print("Duplicados na Natação:", duplicados_natacao)
print("Duplicados no Ciclismo:", duplicados_ciclismo)
print("Duplicados na Corrida:", duplicados_corrida)

#removendo os duplicados
set_natacao = list(set(natacao))
set_ciclismo = list(set(ciclismo))
set_corrida = list(set(corrida))

print(set_natacao)
print(set_ciclismo)
print(set_corrida)

#verificando se os duplicados foram removidos
def verificar_duplicados(lista):
    duplicados = {}
    for aluno in lista:
        if lista.count(aluno) > 1:
            duplicados[aluno] = lista.count(aluno)
    return duplicados

duplicados_natacao = verificar_duplicados(natacao)
duplicados_ciclismo = verificar_duplicados(ciclismo)
duplicados_corrida = verificar_duplicados(corrida)

print("Duplicados na Natação:", duplicados_natacao)
print("Duplicados no Ciclismo:", duplicados_ciclismo)
print("Duplicados na Corrida:", duplicados_corrida)

Duplicados na Natação: {'Marcos Santos': 2, 'Luciana Barbosa': 2}
Duplicados no Ciclismo: {'Marcos Santos': 2}
Duplicados na Corrida: {'Marcos Santos': 2, 'Luciana Barbosa': 2}
['Marcela Rangel', 'Carlos Antunes', 'Joana Ramos', 'Ana Schimidt', 'Marcos Santos', 'Luciana Barbosa', 'Nicolas da Silva', 'André Consentino']
['Marcela Rangel', 'Luis Pereira', 'Ana Schimidt', 'Marcos Santos', 'Michele Souza', 'Marcelo Bach', 'Luciana Barbosa', 'Nicolas da Silva']
['Marcela Rangel', 'Luis Gomes', 'Luis Pereira', 'Joana Ramos', 'Ana Schimidt', 'Marcos Santos', 'Luciana Barbosa', 'André Consentino']
Duplicados na Natação: {'Marcos Santos': 2, 'Luciana Barbosa': 2}
Duplicados no Ciclismo: {'Marcos Santos': 2}
Duplicados na Corrida: {'Marcos Santos': 2, 'Luciana Barbosa': 2}


In [96]:
#14.2. Em seguida, contabilize a quantidade de alunos inscritos em cada modalidade e imprima os três valores

#contando os alunos
qtd_natacao = len(set_natacao)
qtd_ciclismo = len(set_ciclismo)
qtd_corrida = len(set_corrida)

print(qtd_natacao)
print(qtd_ciclismo)
print(qtd_corrida)

8
8
8


In [97]:
#14.3 Imprima uma listagem com o nome de todos os alunos nas 3 modalidades e a quantidade total de alunos, somadas as 3 modalidades

#alunos nas 3 modalidades
#alunos_3 = natacao & ciclismo & corrida
#print(alunos_3)

alunos_3 = natacao + ciclismo + corrida
#print(alunos_3)
set_alunos_3 = list(set(alunos_3))
#print(set_alunos_3)
qtd_alunos_3 = len(set_alunos_3)
#print(qtd_alunos_3)

print(f'A quantidade total de alunos nas 3 modalidades são {qtd_alunos_3} e os alunos são {set_alunos_3}')

A quantidade total de alunos nas 3 modalidades são 12 e os alunos são ['Marcela Rangel', 'Luis Gomes', 'Carlos Antunes', 'Luis Pereira', 'Joana Ramos', 'Ana Schimidt', 'Marcos Santos', 'Michele Souza', 'Marcelo Bach', 'Luciana Barbosa', 'Nicolas da Silva', 'André Consentino']


In [119]:
#14.4 Imprima uma listagem com o nome de cada aluno e as modalidades em que está inscrito

inscricoes = {
    'NATACAO': natacao,
    'CICLISMO': ciclismo,
    'CORRIDA': corrida
}

print('Listagem de alunos e suas modalidades')
for modalidade, alunos in inscricoes.items():
  print('\n')
  print(modalidade)
  for aluno in alunos:
    print('-', aluno)

Listagem de alunos e suas modalidades


NATACAO
- Nicolas da Silva
- Marcos Santos
- Luciana Barbosa
- Ana Schimidt
- Marcos Santos
- Joana Ramos
- André Consentino
- Marcela Rangel
- Luciana Barbosa
- Carlos Antunes


CICLISMO
- Nicolas da Silva
- Marcos Santos
- Luciana Barbosa
- Ana Schimidt
- Marcos Santos
- Marcela Rangel
- Michele Souza
- Marcelo Bach
- Luis Pereira


CORRIDA
- Marcos Santos
- Luciana Barbosa
- Ana Schimidt
- Marcos Santos
- Luis Pereira
- Joana Ramos
- André Consentino
- Marcela Rangel
- Luis Gomes
- Luciana Barbosa


In [118]:
#14.5 Imprima uma listagem com os alunos que se inscreveram em apenas 1 ou em 3 modalidades

alunos_total = natacao + ciclismo + corrida
aluno_modalidades = {}

for aluno in alunos_total:
  qtd_modalidades = 0
  if aluno in natacao:
    qtd_modalidades +=1
  if aluno in ciclismo:
    qtd_modalidades +=1
  if aluno in corrida:
    qtd_modalidades +=1
  aluno_modalidades[aluno] = qtd_modalidades

print('Alunos que se inscreveram em apenas 1 ou 3 modalidades\n')
for aluno, qtd_modalidades in aluno_modalidades.items():
  if qtd_modalidades == 1 or qtd_modalidades == 3:
    print(aluno)

Alunos que se inscreveram em apenas 1 ou 3 modalidades

Marcos Santos
Luciana Barbosa
Ana Schimidt
Marcela Rangel
Carlos Antunes
Michele Souza
Marcelo Bach
Luis Gomes


In [117]:
#14.5 Imprima uma listagem com os alunos que se inscreveram em apenas 1 ou em 3 modalidades

#criando dic
aluno_modalidade = {}

for aluno in natacao:
  aluno_modalidade[aluno] = ['natacao']

for aluno in ciclismo:
  if aluno in aluno_modalidade:
    aluno_modalidade[aluno].append('ciclismo')
  else:
    aluno_modalidade[aluno] = ['ciclismo']

for aluno in corrida:
  if aluno in aluno_modalidade:
    aluno_modalidade[aluno].append('corrida')
  else:
    aluno_modalidade[aluno] = ['corrida']

print("Alunos que se inscreveram em apenas 1 ou 3 modalidades\n")
for aluno, modalidade in aluno_modalidade.items():
    if len(modalidade) == 1 or len(modalidade) == 3:
        print(f"{aluno}: {', '.join(modalidade)}")

Alunos que se inscreveram em apenas 1 ou 3 modalidades

Ana Schimidt: natacao, ciclismo, corrida
Marcela Rangel: natacao, ciclismo, corrida
Carlos Antunes: natacao
Michele Souza: ciclismo
Marcelo Bach: ciclismo
Luis Gomes: corrida


15) Abaixo é apresentado um dicionário contendo os seguintes dados de alguns vencedores do Prêmio Nobel da Paz:
{Ano da premiação: (Nome, País de Origem, Ano de Nascimento, Motivação do prêmio, gênero (Feminino, Masculino ou Organização))}

Faça o que se pede nos subtitens 15.1 e 15.2

In [101]:
nobel = {2011: [['Ellen Jonhson Sirleaf', 'Liberia', 1938, 'woman rights', "F"],['Leymah Gbowee', 'Liberia', 1972, 'woman rights', 'F'], ['Tawakkol', 'Liberia', 1979, 'woman rights', 'F' ]],
         2012: ['Eurpean Union', 'Europe', 1952, 'human rights','O'],
         2013: ['Organization for the Prohibition of Chemical Weapons', 'The Netherlands', 1997, 'eliminate chemical weapons', 'O'],
         2014: [['Kailash Satyarthi', 'India', 1954, 'children education', 'M'], ['Malala Yousafzai', 'Pakistan', 1997, 'children education', 'O']],
         2015: ['National Dialogue Quartet', 'Tunisia', 2011, 'democracy', 'O'],
         2016: ['Juan Manuel Santos', 'Colombia', 1951, 'end of civil war', 'O'],
         2017: ['International Campaign to Abolish Nuclear Weapons', 'Australia', 2007, 'nuclear weapons prohibition', 'O'],
         2018: ['Denis Mukwege', 'Congo', 1955, 'end sexual violation at war', 'M'],
         2019: ['Abiy Ahmed Ali', 'Ethiopia', 1976, 'international conflicts', 'M'],
         2020: ['World Food Programme', 'Italy', 1961, 'combat hunger', 'O']}

15.1) Quais foram os vencedores do Prêmio Nobel da Paz em 2011?

In [148]:
print("Vencedores do Prêmio Nobel da Paz em 2011\n")

for vencedor in nobel[2011]:
  print('2011', vencedor[0])

Vencedores do Prêmio Nobel da Paz em 2011

2011 Ellen Jonhson Sirleaf
2011 Leymah Gbowee
2011 Tawakkol


15.2) Imprima a relação dos países de origem dos laureados com o Prêmio Nobel da Paz na última década.

In [149]:
print("Países de origem dos laureados na última década\n")

#ano atual
import datetime
ano_atual = datetime.datetime.now().year
#print(ano_atual)

#ultima decada
ult_decada = range(ano_atual - 10, ano_atual)

#paises
pais = set()
paises_ano = {}

for ano in ult_decada:
  if ano in nobel:
    vencedores = nobel[ano]
    for vencedor in vencedores:
      if isinstance(vencedor, list):
        pais = vencedor[1]
      else:
        pais = vencedores[1]
      paises_ano[pais] = ano

for pais, ano in paises_ano.items():
  print(ano, pais)

Países de origem dos laureados na última década

2013 The Netherlands
2014 India
2014 Pakistan
2015 Tunisia
2016 Colombia
2017 Australia
2018 Congo
2019 Ethiopia
2020 Italy


16) Em uma competição de salto em distância cada atleta tem direito a três saltos. O resultado do atleta será determinado pela média dos três valores saltos.

Você deve fazer um programa que receba:

* o nome e as três distâncias alcançadas pelo atleta em seus saltos (via input())
* informe o nome, os saltos e a média dos saltos.
* O programa deve ser encerrado quando não for informado o nome do atleta.
* A saída do programa deve ser conforme o exemplo abaixo:

Atleta: Rodrigo

Primeiro Salto: 3 m

Segundo Salto: 2 m

Terceiro Salto: 4 m

Resultado final:

Atleta: Rodrigo

Saltos: 3 - 2 - 4

Média dos saltos: 3 m


17) Qual é a diferença entre o **while** e **for**?

RESPOSTA:

18) Faça um Programa que peça a idade e a altura de 3 pessoas, armazene cada informação na sua respectiva lista e imprima a idade e a altura de cada pessoa.

19) Dados os dois vetores abaixo (listas), faça um programa que leia os dois vetores e gere um terceiro vetor de 20 elementos, cujos valores deverão ser compostos pelos elementos intercalados dos dois outros vetores.

In [102]:
a = [0,2,4,6,8,10,12,14,16,18]
b = [1,3,5,7,9,11,13,15,17,19]

20) Faça UMA FUNÇÃO que recebe um valor n e que imprima:
  
    1
  
    2   2
  
    3   3   3
  
    .....
  
    n   n   n   n   n   n  ... n

21) Faça uma FUNÇÃO que recebe um valor n e que imprima:
  
  1
  
  1   2
  
  1   2   3
  
  .....
  
  1   2   3   ...  n

22) Faça uma FUNÇÃO que retorne o reverso de um número inteiro passado por parâmetro. Por exemplo: 127 -> 721.

23) Faça uma função que receba dois vetores com 5 elementos aleatórios e inteiros cada. Sua função deverá retornar um terceiro vetor de 20 elementos, cujos valores deverão ser compostos pelos elementos concatenados dos dois outros vetores.

24) Faça uma função que receba uma string e retorne a quantidade de consoantes que existem nela.

25) Altere o exercício 24 para que o retorno seja um dicionário, onde a chaves (keys) são as consoantes e os valores (values) são a quantidade de vezes em que a consoante aparece na string